File: Python/Program/OS.py
from Software import Software

class OS(Software):
    def __init__(self, arsitektur="", kernel="", nama="", versi="", developer=""):
        super().__init__(nama, versi, developer)
        self.__arsitektur = arsitektur
        self.__kernel = kernel
    
    def set_arsitektur(self, arsitektur):
        self.__arsitektur = arsitektur
    
    def set_kernel(self, kernel):
        self.__kernel = kernel
    
    def get_arsitektur(self):
        return self.__arsitektur
    
    def get_kernel(self):
        return self.__kernel

File: CPP/Program/OS.cpp
#ifndef OS_CPP
#define OS_CPP

#include <iostream>
#include <string>
#include "Software.cpp"
using namespace std;

class OS : public Software {
private:
    string arsitektur; // 32-bit, 64-bit
    string kernel;
    
public:
    OS() {
    }
    
    OS(string arsitektur, string kernel, string nama, string versi, string developer) : Software(nama, versi, developer) {
        this->arsitektur = arsitektur;
        this->kernel = kernel;
    }
    
    void setArsitektur(string arsitektur) {
        this->arsitektur = arsitektur;
    }
    
    void setKernel(string kernel) {
        this->kernel = kernel;
    }
    
    string getArsitektur() {
        return this->arsitektur;
    }
    
    string getKernel() {
        return this->kernel;
    }
    
    ~OS() {
    }
};

#endif // OS_CPP